### PR TITLE
Add more queryFactory constructors

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

File: querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java [hunk 1]
```diff
@@ -37,6 +37,20 @@
...
+    }
+
+    public HibernateQueryFactory(JPQLTemplates templates, final Session session) {
+        this.session = new Provider<Session>() {
+            @Override
+            public Session get() {
+                return session;
+            }
+        };
+        this.templates = templates;
```
Comment (johnktims) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java]: You could also create the `Provider<Session>` and pass it to the existing constructor `this(templates, new Provider<Session>...);`.

Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java]: Syntactically it feels cleaner this way.

Comment (johnktims) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java]: That's fine with me. I think this PR is ready to be merged then. @Shredder121 ?

Comment (Shredder121) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java]: Yes, syntactically it is much better this way.
I think it's good to merge, I'll do so.

**Pull request merged**